Lib/asyncore.py

```diff
@@ -127,7 +127,8 @@ def poll(timeout=0.0, map=None):
         map = socket_map
     if map:
         r = []; w = []; e = []
-        for fd, obj in list(map.items()):
+        map_copy = dict(map)
```

Comment:
You can use threads to send and receive from file descriptors watched by asyncore
dispatchers, and this is how you can test asyncore. But you cannot call dispatcher
methods from another thread, or modify asyncore map from other threads. asyncore
does not have builtin support for threading.

To use threads with asyncore you will have to extend it to have a queue of functions
scheduled to run when the event loop wakes up, and a way to wake up the event loop, 
and a new asyncore.loop function integrating these features. Such code exists in the
old meduza project:
http://www.nightmare.com/medusa/
 
If you are using python 2.7, you may find this useful - backport of python 3 event loop
to 2.7 and integration with asyncore. This add threading and timeouts support.
https://github.com/oVirt/vdsm/blob/master/lib/vdsm/storage/asyncevent.py

If you are using python 3, I think it is better to use asyncio, it is very hard to get error
handling right with asyncore.